docs/developers/config_tin.py

```diff
@@ -0,0 +1,76 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+from enum import Enum
+
+
+class ArmoryMode(Enum):
+    HOST = "host"
+    DOCKER = "docker"
+    OFFLINE = "offline"
+
+
+@dataclass
+class ArmoryCredentials:
+    github_token: str
+    s3_id: str
+    s3_key: str
+
+
+@dataclass
+class ArmoryPaths:
+    output_dir: Path
+    local_git_dir: Path
+
+
+def paths_by_mode(mode: str) -> ArmoryPaths:
```

Comment:
I fundamentally don't like this pattern.  I feel like the paths should be independent of mode.  IF running in docker, then armory should be setup in docker with appropriate paths / config and then armory just does what it does.  If we want to map directories from docker to host machine, then we do that and redo the config to make it appropriate.